src/stories/containers/recognized-delegates/useRecognizedDelegates.mvvm.tsx

```diff
@@ -96,6 +95,66 @@ const useRecognizedDelegates = (delegates: DelegatesDto) => {
       onPrevious,
     });
 
+  useEffect(() => {
+    // change the tabs when anchor changes
+    if (anchor) {
+      const index = Object.values(DELEGATES_IDS_ENUM).findIndex((id) => anchor.indexOf(id) > -1);
+      if (index !== -1) {
+        const indexKey = Object.keys(DELEGATES_IDS_ENUM)[index];
+
+        setTabsIndex(DELEGATES_IDS_ENUM[indexKey as keyof typeof DELEGATES_IDS_ENUM]);
+        setTabsIndexNumber(index);
+      }
+    }
+  }, [anchor]);
+  const wallets: BudgetStatementWalletDto[] = useMemo(() => {
+    const dict: { [id: string]: BudgetStatementWalletDto } = {};
+    if (!delegates.budgetStatements) return [];
+    const budgetStatement = delegates.budgetStatements?.find(
+      (bs) => bs.month === currentMonth.toFormat(API_MONTH_TO_FORMAT)
+    );
```

Comment:
This is not necessary as you can access to the `currentBudgetStatement` that is the budget statement selected for the current month so all the operations should be using that budget statement